testservices/neutronmodel/neutronmodel.go

```diff
@@ -320,8 +321,25 @@ func (n *NeutronModel) AddSecurityGroupRule(ruleId string, rule neutron.RuleInfo
 	if rule.IPProtocol != "" {
 		newrule.IPProtocol = &rule.IPProtocol
 	}
-	if rule.EthernetType != "" {
-		newrule.EthernetType = rule.EthernetType
+	if rule.EthernetType == "" || rule.EthernetType == "IPv4" || rule.EthernetType == "IPv6" {
+		if (rule.EthernetType == "") {
+			// Neutron assumes IPv4 if no EthernetType is specified
+			newrule.EthernetType = "IPv4"
+		} else {
+			newrule.EthernetType = rule.EthernetType
+		}
+	} else {
+		return testservices.NewSecurityGroupRuleInvalidEthernetType(rule.EthernetType)
+	}
```

Comment:
Just a personal readability opinion; this might be clearer:
```
        switch rule.EthernetType {
        case "": 
                // Neutron assumes IPv4 if no EthernetType is specified
                newrule.EthernetType = "IPv4"
        case "IPv4", "IPv6":
                newrule.EthernetType = rule.EthernetType
        default:
                return testservices.NewSecurityGroupRuleInvalidEthernetType(rule.EthernetType)
        }
```